crates/matrix-sdk-ui/src/timeline/controller/state.rs

```diff
@@ -536,24 +542,30 @@ impl TimelineStateTransaction<'_> {
                         timestamp: Some(event.origin_server_ts()),
                         visible: false,
                     };
-                    let _event_added_or_updated = self
-                        .add_or_update_remote_event(
-                            event_meta,
-                            position,
-                            room_data_provider,
-                            settings,
-                        )
-                        .await;
+
+                    // Remember the event before returning prematurely.
```

Comment:
Perhaps telling us why we're remembering things would be useful as well, not particularly needed since you can follow the call tree of `add_or_update_remote_event()` yourself, but would avoid a couple of raised eyebrows.

Reply:
Good idea. I'll improve the doc.